pinot-core/src/test/java/org/apache/pinot/queries/VarianceQueriesTest.java

```diff
@@ -0,0 +1,446 @@
+    for (int i = 0; i < NUM_RECORDS; i++) {
+      expectedStdDevs[0].increment(_intValues[i]);
+      expectedStdDevs[1].increment(_longValues[i]);
+      expectedStdDevs[2].increment(_floatValues[i]);
+      expectedStdDevs[3].increment(_doubleValues[i]);
+      expectedStdDevs[4].increment(_intValues[i]);
+      expectedStdDevs[5].increment(_longValues[i]);
+      expectedStdDevs[6].increment(_floatValues[i]);
+      expectedStdDevs[7].increment(_doubleValues[i]);
+    }
+
+    double expectedIntSum = Arrays.stream(_intValues).asDoubleStream().sum();
+    double expectedLongSum = Arrays.stream(_longValues).asDoubleStream().sum();
+    double expectedFloatSum = 0.0;
+    for (int i = 0; i < _floatValues.length; i++) {
+      expectedFloatSum += _floatValues[i];
+    }
+    double expectedDoubleSum = Arrays.stream(_doubleValues).sum();
+
+    // Compute the query
+    String query =
+        "SELECT STDDEV_POP(intColumn), STDDEV_POP(longColumn), STDDEV_POP(floatColumn), STDDEV_POP(doubleColumn),"
+            + "STDDEV_SAMP(intColumn), STDDEV_SAMP(longColumn), STDDEV_SAMP(floatColumn), STDDEV_SAMP(doubleColumn) "
+            + "FROM testTable";
+    AggregationOperator aggregationOperator = getOperator(query);
+    AggregationResultsBlock resultsBlock = aggregationOperator.nextBlock();
+    QueriesTestUtils.testInnerSegmentExecutionStatistics(aggregationOperator.getExecutionStatistics(), NUM_RECORDS, 0,
+        NUM_RECORDS * 4, NUM_RECORDS);
+    List<Object> aggregationResult = resultsBlock.getResults();
+
+    // Validate the aggregation results
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(0), NUM_RECORDS, expectedIntSum,
+        expectedStdDevs[0].getResult(), false);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(1), NUM_RECORDS, expectedLongSum,
+        expectedStdDevs[1].getResult(), false);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(2), NUM_RECORDS, expectedFloatSum,
+        expectedStdDevs[2].getResult(), false);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(3), NUM_RECORDS, expectedDoubleSum,
+        expectedStdDevs[3].getResult(), false);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(4), NUM_RECORDS, expectedIntSum,
+        expectedStdDevs[4].getResult(), true);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(5), NUM_RECORDS, expectedLongSum,
+        expectedStdDevs[5].getResult(), true);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(6), NUM_RECORDS, expectedFloatSum,
+        expectedStdDevs[6].getResult(), true);
+    checkWithPrecisionForStandardDeviation((VarianceTuple) aggregationResult.get(7), NUM_RECORDS, expectedDoubleSum,
+        expectedStdDevs[7].getResult(), true);
+
+    // Update the expected result by 3 more times (broker query will compute 4 identical segments)
```

Comment:
The variance on 4 identical segments is the same as variance on one of these segments. Should probably to consider using `getDistinctInstances` in `BaseQueriesTest`

Reply:
@jasperjiaguo Can you elaborate on your suggestion?

I referred covariance test and it also uses `getBrokerResponse()` to get the result. According to the documentation, `getBrokerResponse()` seems to compute the result from 4 identical segments.  FYI,`getBrokerResponse()` internally calls `getDistinctInstances()`.

Reply:
Yes, in `CovarianceQueriesTest`, @SabrinaZhaozyf has added the [getDistinctInstances()](https://github.com/apache/pinot/blob/270315a13f457376df04cd47bc62f84d1c1e558e/pinot-core/src/test/java/org/apache/pinot/queries/CovarianceQueriesTest.java#L157) to use 4 distinct segments for these statistical functions because Covar on 4 identical segments is the same as Covar on one of each segment, as kind of defeats the purpose of testing merge logic over multiple segments. The `getDistinctInstances` in base class is still returning 4 identical segments for backward compat. We can probably inherit `CovarianceQueriesTest` in `VarianceQueriesTest`.

Reply:
LGTM, thanks!